xarray/backends/plugins.py

```diff
@@ -113,10 +113,22 @@ def guess_engine(store_spec):
 
 
 def get_backend(engine):
-    """Select open_dataset method based on current engine"""
-    engines = list_engines()
-    if engine not in engines:
-        raise ValueError(
-            f"unrecognized engine {engine} must be one of: {list(engines)}"
+    """Select open_dataset method based on current engine."""
+    if isinstance(engine, str):
+        engines = list_engines()
+        if engine not in engines:
+            raise ValueError(
+                f"unrecognized engine {engine} must be one of: {list(engines)}"
+            )
+        backend = engines[engine]
+    elif inspect.isclass(engine) and issubclass(engine, BackendEntrypoint):
```

Comment:
Nit: I would just use `isinstance(..., type)` here (which turns out to be exactly how `inspect.isclass` is implemented):
```suggestion
    elif isinstance(engine, type) and issubclass(engine, BackendEntrypoint):
```